src/Index.php

```diff
@@ -164,9 +166,11 @@ public function updateDocuments(array $docs, array $options = []): ResponseSet
     /**
      * Update entries in the db based on a query.
      *
-     * @param AbstractQuery|array|Collapse|Query|string|Suggest $query   Query object or array
-     * @param AbstractScript                                    $script  Script
-     * @param array                                             $options Optional params
+     * @param AbstractQuery|AbstractSuggest|array|Collapse|Query|string|Suggest|null $query Query object or array
```

Comment:
This PR looks nice! I was updating these lines recently to add `AbstractSuggest`, but in this case I remember I didn't add it because I wasn't sure if that makes sense.

`Query::create()` accepts all these types, but in this particular case, `updateByQuery` (and `deleteByQuery`) I guess `AbstractSuggest`, `Collapse` and `Suggest` doesn't make sense here and probably `null` as well.

Reply:
Yeah, that makes sense. I removed abstract/suggest and collapse. I left null in place because it's used to create a match all query.